### PR TITLE
Updating docker version in ARM template to use latest release tagged image

### LINKED ISSUE #666: [FR] Change docker publish flow to use release tag instead of latest tag
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### Feature Request Proposal

Docker image publishing process change proposal

- For release branches, automatically publish docker image with corresponding release tag, e.g, feathr-registry:releases-v0.7.2
- Ensure deployment picks up latest released docker image, seems the easilest way to do this is link latest tag to LKG release tag.

### Motivation

> #### What is the use case for this feature?

No impact to end user experience. Only impacts deployment experience, this ask ensures correct registry docker image is used for deployment.

### Details

_No response_

### What component(s) does this feature request affect?

- [ ] `Python Client`: This is the client users use to interact with most of our API. Mostly written in Python.
- [ ] `Computation Engine`: The computation engine that execute the actual feature join and generation work. Mostly in Scala and Spark.
- [ ] `Feature Registry API`: The frontend API layer supports SQL, Purview(Atlas) as storage. The API layer is in Python(FAST API)
- [ ] `Feature Registry Web UI`: The Web UI for feature registry. Written in React

### PR DESCRIPTION
**Pull request by jainr**
## Description
Changing the deployment template to pull latest released docker API/registry image
The current behavior is to pull the latest which we don't update as part of CD.
Resolves #666 

## How was this PR tested?
Doing a fresh deployment.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

**Review by xiaoyongzhu — APPROVED** [review]

File: docs/dev_guide/feathr_overall_release_guide.md [hunk 1]
```diff
@@ -30,11 +30,12 @@ Read through the [commit log](https://github.com/feathr-ai/feathr/commits/main)
 
 ## Code Changes
 Before the release is made, the version needs to be updated in following places
-- [build.sbt](https://github.com/feathr-ai/feathr/blob/main/build.sbt#L3) - For Maven release version
-- [setup.py](https://github.com/feathr-ai/feathr/blob/main/feathr_project/setup.py#L10) - For PyPi release version
-- [conf.py](https://github.com/feathr-ai/feathr/blob/main/feathr_project/docs/conf.py#L27) - For documentation version
+- [build.sbt](https://github.com/linkedin/feathr/blob/main/build.sbt#L3) - For Maven release version
```
Comment (blrchen) [inline docs/dev_guide/feathr_overall_release_guide.md:33]: The org name has been changed from LinkedIn to feathr-ai, this change should get reverted

**Comment by blrchen** [comment]
Thanks for the PR, could you please also update the `Code Changes` section in https://github.com/feathr-ai/feathr/blob/main/docs/dev_guide/feathr_overall_release_guide.md to mention update image tag in arm deployment json as well

File: docs/dev_guide/feathr_overall_release_guide.md (hunk 1 above)
Comment (jainr) [reply docs/dev_guide/feathr_overall_release_guide.md]: This has been fixed.

**Comment by jainr** [comment]
I did update the code change section to point out to release engineer about this change.

**Review by blrchen — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**